src/components/InngestStepTools.test.ts

```diff
@@ -0,0 +1,197 @@
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import { StepOpCode } from "../types";
+import { createStepTools, StepFlowInterrupt } from "./InngestStepTools";
+
+describe("waitForEvent", () => {
+  let waitForEvent: ReturnType<typeof createStepTools>[0]["waitForEvent"];
+  let state: ReturnType<typeof createStepTools>[1];
+
+  beforeEach(() => {
+    [{ waitForEvent }, state] = createStepTools({});
+  });
+
+  test("return WaitForEvent step op code", async () => {
+    expect(() => waitForEvent("event")).toThrow(StepFlowInterrupt);
+    await expect(state.nextOp).resolves.toMatchObject({
+      op: StepOpCode.WaitForEvent,
+    });
+  });
+
+  test("returns `event` as ID", async () => {
+    expect(() => waitForEvent("event")).toThrow(StepFlowInterrupt);
+    await expect(state.nextOp).resolves.toMatchObject({
+      id: "event",
+    });
+  });
+
+  test("return blank opts if none given", async () => {
+    expect(() => waitForEvent("event")).toThrow(StepFlowInterrupt);
+    await expect(state.nextOp).resolves.toMatchObject({
+      opts: {},
+    });
+  });
+
+  test("return a hash of the op", async () => {
+    expect(() => waitForEvent("event")).toThrow(StepFlowInterrupt);
+    await expect(state.nextOp).resolves.toMatchObject({
+      hash: "17ffb6f7b5118797c77057e5176fdc2412e275fe",
+    });
+  });
+
+  test("return TTL if string `timeout` given", async () => {
+    expect(() => waitForEvent("event", { timeout: "1m" })).toThrow(
+      StepFlowInterrupt
+    );
+    await expect(state.nextOp).resolves.toMatchObject({
+      opts: {
+        ttl: "1m",
+      },
+    });
+  });
+
+  test("return TTL if date `timeout` given", async () => {
+    const upcoming = new Date();
+    upcoming.setDate(upcoming.getDate() + 6);
+    upcoming.setHours(upcoming.getHours() + 1);
+
+    expect(() => waitForEvent("event", { timeout: upcoming })).toThrow(
+      StepFlowInterrupt
+    );
+    await expect(state.nextOp).resolves.toMatchObject({
+      opts: {
+        ttl: expect.stringContaining("6d"),
+      },
+    });
+  });
+
+  test("return simple field match if `match` string given", async () => {
+    expect(() => waitForEvent("event", { match: "name" })).toThrow(
+      StepFlowInterrupt
+    );
+    await expect(state.nextOp).resolves.toMatchObject({
+      opts: {
+        match: "event.name == async.name",
+      },
+      hash: "b3702cc0450f2e014e1ea9b9d2438e8f3127c95a",
+    });
+  });
+
+  test("return custom match statement if `if` given", async () => {
+    expect(() => waitForEvent("event", { if: "name == 123" })).toThrow(
+      StepFlowInterrupt
+    );
+    await expect(state.nextOp).resolves.toMatchObject({
+      opts: {
+        match: "name == 123",
+      },
+      hash: "9d3fe213c82f60f52e91b036388fa7851b7ef6b3",
+    });
+  });
+
+  test("prioritise `match` statement if both `match` and `if` given", async () => {
+    expect(() =>
+      waitForEvent("event", { match: "name", if: "name == 123" })
```

Comment:
@djfarrelly Not sure what you mean? Do you mean if the opts for `waitForEvent` are incorrect?

Reply:
Yeah @jpwilliams, exactly. If we're ignoring one of the options, should we throw and error that says something like "Can only pass match or if, but not both, defaulting to match?" I'm just thinking how to share feedback to the user since they may not realize in this case that their `if` is being superseded by the `match`

Reply:
I'll make the opts so it has to be either one or the other, but not both.

Reply:
![image](https://user-images.githubusercontent.com/1736957/200813334-aec43ed1-3e7a-4c7a-aaf0-78e46a7c19d1.png)
![image](https://user-images.githubusercontent.com/1736957/200813361-bc0167bc-cbdf-4292-8afe-b1753c59d9f1.png)
![image](https://user-images.githubusercontent.com/1736957/200813451-9ee8873b-0a17-4867-acdb-f9da4bc1aff6.png)